### PR TITLE
First draft of dataloader.

### PR DESCRIPTION
**Pull request by j-simpson1**
- added dataloader using libraries in the PyTorch ecosystem (simple_dataloader), which you can access image-level labels.
- I have also added a custom dataloader (custom_data and custom_loader), which allows you to access species, breed and bounding boxes.

### CODE REVIEW

File: dataloader.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+OxfordPets_dataset = ConcatDataset(full_dataset)
+
+# Splitting the dataset: 70% training, 15% validation, 15% testing
+total_size = len(OxfordPets_dataset)
+train_size = int(0.7 * total_size)
+val_size = int(0.15 * total_size)
+test_size = total_size - train_size - val_size
+
+generator = torch.Generator().manual_seed(2025)
```
Comment (davidve0206) [inline dataloader.py:38]: Do we need to seed again here? (as there is already a torch.manual_seed at the top)?

Comment (j-simpson1) [reply dataloader.py]: David!! I hope you're well. I just removed the extra seed, so we only have one!

**Review by timfarkas — APPROVED** [review]
